### PR TITLE
Update KNMI methodology

### PR DESCRIPTION
**Pull request by martinvonk**

### CODE REVIEW

**Comment by OnnoEbbens** [comment]
Thanks for this PR, this should speed up the 'fill_missing_measurements' a lot.

**Review by OnnoEbbens — CHANGES_REQUESTED** [review]

File: hydropandas/io/knmi.py [hunk 1]
```diff
@@ -1006,40 +1082,26 @@ def read_knmi_daily_rainfall(f, meteo_var):
     return df, variables
 
 
-def _read_station_location(f):
-    stations = None
+def get_knmi_api(url: str, params: Dict[str, str]) -> StringIO:
```
Comment (OnnoEbbens) [inline hydropandas/io/knmi.py:1076]: missing docstring

**Review by OnnoEbbens — CHANGES_REQUESTED** [review]
Can you add and fix some of the docstrings? Then this PR is ready to be merged

**Comment by OnnoEbbens** [comment]
Oh oh I celebrated to early. There is still an error in the knmi notebook.

File: tests/test_001_to_from.py [hunk 2]
```diff
@@ -245,14 +245,14 @@ def test_from_excel():
 
 def test_pressure_obs_from_stn():
     hpd.MeteoObs.from_knmi(
-        stn=310, meteo_var="P", interval="hourly", start="2022-1-1", end="2023-1-1"
+        stn=310, meteo_var="PG", interval="hourly", start="2022-1-1", end="2023-1-1"
```
[the remark this answers is not in the document]
Comment (martinvonk) [reply tests/test_001_to_from.py]: mistake on my part, didn't notice the hourly interval.

File: hydropandas/io/knmi.py [hunk 3]
```diff
@@ -1142,127 +1184,144 @@ def get_knmi_daily_meteo_url(stn, meteo_var, start=None, end=None):
...
+    df: DataFrame
+        dataframe with meteo_var as column
+    meta: dictionary
+        dictionary with meteo_var as key
+    meteo_var : str
+        e.g. 'EV24'.
+    start : pd.TimeStamp or None
+        start time of observations.
+    end : pd.TimeStamp or None
+        end time of observations.
```
[the remark this answers is not in the document]
Comment (martinvonk) [reply hydropandas/io/knmi.py:1134]: added

File: hydropandas/io/knmi.py (hunk 1 above)
Comment (martinvonk) [reply hydropandas/io/knmi.py]: added the docstring

**Review by OnnoEbbens — CHANGES_REQUESTED** [review]
I think we should keep the behavior to return an empty dataframe when there is no data (instead of raising a pretty vague error). Is that doable?

**Comment by martinvonk** [comment]
I'll have a look at the different kind's of exceptions raised when downloading KNMI data.

**Review by OnnoEbbens — APPROVED** [review]

**Pull request merged**